tests/Unit/Core/FeatureFlag/Layer/QueryLayerTest.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace Tests\Unit\Core\FeatureFlag\Layer;
+
+use Generator;
+use InvalidArgumentException;
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Adapter\Environment;
+use PrestaShop\PrestaShop\Core\EnvironmentInterface;
+use PrestaShop\PrestaShop\Core\FeatureFlag\Layer\QueryLayer;
+use Symfony\Component\HttpFoundation\Request;
+
+class QueryLayerTest extends TestCase
+{
+    private const FEATURE_FLAG_TEST = 'feature_flag_test';
+    private const VAR_FEATURE_FLAG_TEST = 'PS_FF_FEATURE_FLAG_TEST';
+
+    public static EnvironmentInterface $environment;
+    public static Request $request;
+
+    public function setUp(): void
+    {
+        $this->resetEnv();
+    }
+
+    private function resetEnv(): void
+    {
+        $this->modeDev(true);
+        static::$request = new Request();
+    }
+
+    private function modeDev(bool $status): void
+    {
+        static::$environment = new Environment($status, 'unit');
```

Comment:
You should use some newly constructed mocks in each test functions instead of relying on static fields